packages/wallets/src/features/cashier/flows/WalletWithdrawal/WalletWithdrawal.tsx

```diff
@@ -1,25 +1,35 @@
 import React, { useEffect, useState } from 'react';
-import { useActiveWalletAccount, useCurrencyConfig } from '@deriv/api';
+import { useActiveWalletAccount, useAuthorize, useCurrencyConfig } from '@deriv/api';
 import { Loader } from '../../../../components';
 import { WithdrawalCryptoModule, WithdrawalFiatModule, WithdrawalVerificationModule } from '../../modules';
 
 const WalletWithdrawal = () => {
     const { getConfig, isSuccess: isCurrencyConfigSuccess } = useCurrencyConfig();
+    const { switchAccount } = useAuthorize();
     const { data: activeWallet } = useActiveWalletAccount();
     const [verificationCode, setVerificationCode] = useState('');
 
     useEffect(() => {
         const queryParams = new URLSearchParams(location.search);
+        const loginidQueryParam = queryParams.get('loginid');
         const verificationQueryParam = queryParams.get('verification');
 
+        // if loginid query param doesn't match active wallet's loginid on mount, initiate account switching
+        if (loginidQueryParam && loginidQueryParam !== activeWallet?.loginid) {
+            switchAccount(loginidQueryParam);
+            return;
```

Comment:
return <Loader> or smth?

Reply:
Calling `switchAccount` triggers reloading components higher in the component tree, so I skipped using loader in this one, should I use it after all?

Reply:
yeah I guess then its unavoidable :( hope upcoming changes will make it easier to handle such cases